jenkins/databricks/test.sh

```diff
@@ -105,7 +102,7 @@ if [[ "$(pwd)" == "$SOURCE_PATH" ]]; then
 
     if [[ "$TEST_MODE" == "DEFAULT" || $TEST_MODE == "CI_PART2" || "$TEST_MODE" == "DELTA_LAKE_ONLY" ]]; then
         ## Run Delta Lake tests
-        SPARK_SUBMIT_FLAGS="$SPARK_CONF $DELTA_LAKE_CONFS" TEST_PARALLEL=1 \
+        DRIVER_MEMORY="4g" TEST_PARALLEL=8 \
```

Comment:
We should remove `TEST_PARALLEL=1/TEST_PARALLEL=8` here to let the script run_pyspark_from_build.sh calculate the parallel number by its own.

BTW, can you share the comparison of time statics from single tests and multi parallelism for delta_lake tests here? Thanks!